src/ucp/wireup/wireup.c

```diff
@@ -582,26 +692,75 @@ ucp_wireup_process_pre_request(ucp_worker_h worker, ucp_ep_h ep,
 
     status = ucp_ep_config_err_mode_check_mismatch(ep, msg->err_mode);
     if (status != UCS_OK) {
-        goto err_ep_set_failed;
+        ucp_ep_set_failed_schedule(ep, UCP_NULL_LANE, status);
+        return;
     }
 
-    /* restore the EP here to avoid access to incomplete configuration before
-       this point */
-    ucp_ep_update_remote_id(ep, msg->src_ep_id);
+    ucp_wireup_process_pre_flow_msg(ep, remote_address, ep_init_flags,
+                                    msg->src_ep_id);
+}
 
-    /* initialize transport endpoints */
-    status = ucp_wireup_init_lanes(ep, ep_init_flags, &ucp_tl_bitmap_max,
-                                   remote_address, addr_indices,
-                                   &am_need_flush);
+static ucs_status_t
+ucp_wireup_match_ep_by_connection(ucp_worker_h worker,
+                                  const ucp_unpacked_address_t *remote_address,
+                                  ucp_ep_match_conn_sn_t conn_sn,
+                                  unsigned ep_init_flags, ucp_ep_h *ep_p)
+{
+    uint64_t remote_uuid = remote_address->uuid;
+    ucp_ep_h ep;
+    ucs_status_t status;
+
+    ep = ucp_ep_match_retrieve(worker, remote_uuid,
+                               conn_sn ^ (remote_uuid == worker->uuid),
+                               UCS_CONN_MATCH_QUEUE_EXP);
+    if (ep != NULL) {
+        goto out;
+    }
+
+    /* Create a new endpoint if does not exist */
+    status = ucp_ep_create_base(worker, ep_init_flags, remote_address->name,
+                                "remote-request", &ep);
     if (status != UCS_OK) {
-        goto err_ep_set_failed;
+        return status;
     }
 
-    ucp_wireup_send_request(ep);
-    return;
+    /* Add internal endpoint to hash */
+    ep->conn_sn = conn_sn;
+    if (!ucp_ep_match_insert(worker, ep, remote_uuid, ep->conn_sn,
+                             UCS_CONN_MATCH_QUEUE_UNEXP) &&
+        (worker->context->config.features & UCP_FEATURE_STREAM)) {
+        ucs_diag("worker %p: created the endpoint %p without"
+                 " connection matching, but Stream API support was"
+                 " requested on the context %p",
+                 worker, ep, worker->context);
+    }
 
-err_ep_set_failed:
-    ucp_ep_set_failed_schedule(ep, UCP_NULL_LANE, status);
+out:
+    *ep_p = ep;
+    return UCS_OK;
+}
+
+static int ucp_wireup_should_ignore_request(ucp_ep_h ep, uint64_t remote_uuid,
```

Comment:
maybe rename `ucp_wireup_resolve_request`?

Reply:
IMO the purpose of this function is to decide whether to ignore the request or not

Reply:
IMO the term `resolution` can refer to other stuff which is not necessarily relates to head-to-head resolution